### ISSUE TITLE
Off-by-one error in text preprocessing (sequence_to_matrix)

### ISSUE THREAD

**Issue by dtieber** [keras-team/keras-preprocessing#78]
Seems like the num_words property in [text.py](https://github.com/keras-team/keras-preprocessing/blob/4872fccc75af22a123695dbf3b82bdb424372709/keras_preprocessing/text.py#L399) is not initialized with the correct length. I found this out because I'm using this value in order to calculate the number of input/output neurons which leads to issues when I'm training the model.

I think num_words should be initialized like this: `num_words = len(self.word_index)` if not set explicitly.

**Comment by Dref360** [comment]
To avoid breaking changes, we should document this behaviour.

**Comment by Dref360** [comment]
Yup I saw it. We will plan a major release with all off-by-one fixes. Right now, this would cause a lot of projects to break.
